scipy/integrate/_quadpack_py.py

```diff
@@ -435,6 +435,18 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
     if not isinstance(args, tuple):
         args = (args,)
 
+    # if a == b, directly return 0 with error 0
+    if a == b:
+        if full_output == 0:
+            return (0., 0.)
+        else:
+            return (0., 0., {"neval": 0, "last": 0,
+                             "alist": np.full(limit, np.nan, dtype=np.float64),
+                             "blist": np.full(limit, np.nan, dtype=np.float64),
+                             "rlist": np.zeros(limit, dtype=np.float64),
+                             "elist": np.zeros(limit, dtype=np.float64),
+                             "iord" : np.full(limit, np.nan, dtype=np.int32)})
```

Comment:
What do folks think about this implementation of a "dummy" `infodict`? We have to return *something* here, so I tried to make it as close to the form of the dictionary that `quad()` returns under normal circumstances ([see "Notes"](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.quad.html)), without suggesting that the values have meaning that they actually don't. Thoughts?

Reply:
many thanks! Honestly speaking I never used this output option. Since this two-bounds equal case is so special that the implementation of this case should be written separately just like what you did.